x/staking/querier/querier.go

```diff
@@ -129,11 +129,10 @@ func NewQueryRedelegationParams(delegatorAddr sdk.AccAddress, srcValidatorAddr s
 }
 
 func queryValidators(ctx sdk.Context, cdc *codec.Codec, k keep.Keeper) (res []byte, err sdk.Error) {
-	stakingParams := k.GetParams(ctx)
-	validators := k.GetValidators(ctx, stakingParams.MaxValidators)
+	validators := k.GetAllValidators(ctx)
 
 	res, errRes := codec.MarshalJSONIndent(cdc, validators)
-	if err != nil {
+	if errRes != nil {
```

Comment:
Good catch. I recall fixing the same issue elsewhere in staking. Perhaps we should do a quick pass through and make sure there aren't other bugs like this 👍